pkg/api/types.go

```diff
@@ -1305,8 +1305,9 @@ type Endpoints struct {
 //     a: [ 10.10.1.1:8675, 10.10.2.2:8675 ],
 //     b: [ 10.10.1.1:309, 10.10.2.2:309 ]
 type EndpointSubset struct {
-	Addresses []EndpointAddress
-	Ports     []EndpointPort
+	Addresses         []EndpointAddress
+	NotReadyAddresses []EndpointAddress
```

Comment:
Repack subset *should* guarantee that an address doesn't show up in ready /
not ready in the same subset.  I'll update the doc, but I'm not sure what
code is indicating that we can't repack the subset to be unique.

On Wed, Sep 16, 2015 at 9:54 AM, Abhi Shah <notifications@github.com> wrote:

> In pkg/api/types.go
> <https://github.com/kubernetes/kubernetes/pull/13778#discussion_r39631558>
> :
>
> > @@ -1305,8 +1305,9 @@ type Endpoints struct {
> >  //     a: [ 10.10.1.1:8675, 10.10.2.2:8675 ],
> >  //     b: [ 10.10.1.1:309, 10.10.2.2:309 ]
> >  type EndpointSubset struct {
> > -	Addresses []EndpointAddress
> > -	Ports     []EndpointPort
> > +	Addresses         []EndpointAddress
> > +	NotReadyAddresses []EndpointAddress
>
> you seem to split it in the controller code changes,but not in the
> repacksubset code
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13778/files#r39631558>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift